File: benchmarks/vm/13_collatz_loop.py
import time

t0 = time.monotonic()
total_steps = 0
checksum = 0
seed = 1
while seed <= 2000:
    n = seed
    steps = 0
    while n != 1:
        if (n % 2) == 0:
            n //= 2
        else:
            n = (n * 3) + 1
        steps += 1
    total_steps += steps
    checksum += steps * seed
    seed += 1
total_ms = (time.monotonic() - t0) * 1000
print(f"steps = {total_steps}")
print(f"checksum = {checksum}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")

File: benchmarks/vm/13_collatz_loop.lua
local t0 = os.clock()
local total_steps = 0
local checksum = 0
local seed = 1
while seed <= 2000 do
    local n = seed
    local steps = 0
    while n ~= 1 do
        if (n % 2) == 0 then
            n = n // 2
        else
            n = (n * 3) + 1
        end
        steps = steps + 1
    end
    total_steps = total_steps + steps
    checksum = checksum + (steps * seed)
    seed = seed + 1
end
local total_ms = (os.clock() - t0) * 1000
print("steps = " .. tostring(total_steps))
print("checksum = " .. tostring(checksum))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))
